lib/min_heap.rb

```diff
@@ -14,18 +15,28 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(n)
+  # Space Complexity: O(1)
   def add(key, value = key)
```

Comment:
The Binary tree nature of the heap and `heap_up` method mean the time complexity is O(log n) and the recursion causes a space complexity of O(log n) as well.